src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -69,7 +73,11 @@ func (client *Client) intercept(resp http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	defer connOut.Close()
+	defer func() {
+		if err := connOut.Close(); err != nil {
+			log.Debugf("Error closing the out connection: %s", err)
+		}
+	}()
```

Comment:
Yes, that's a solution (passing the sync.once variable). The challenge here is to be able to call the function within the goroutine. What I proposed in the latest commit is to use channel blocking to achieve the same goal without `sync.once` or directly calling, but relying on the defer. Please, let me know what you think of this other possibility :)
https://github.com/getlantern/lantern/blob/6f9e63687a6bf16ff7099f93fe4fae30d4dddc4f/src/github.com/getlantern/flashlight/client/handler.go#L119